docs/dev/connectors/pravega.md

```diff
@@ -0,0 +1,398 @@
+---
+title: "Pravega Connector"
+nav-title: Pravega
+nav-parent_id: connectors
+nav-pos: 6
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+* This will be replaced by the TOC
+{:toc}
+
+## Pravega Introduction
+[Pravega](http://pravega.io/) is an open-source, high-performance, durable, elastic and strongly-consistent
+stream storage for continuous and unbounded data.
+
+Here are some need-to-know Pravega terminologies.
```

Comment:
I suggest to replace this terminology list with the following paragraphs:

```
Pravega builds on the concept of *Stream*. A stream comprises one or more sequences of bytes. 
The unit of a stream is the *segment*. A *segment* is an unbounded sequence of bytes. A stream
can have one or more segments at any time, and the set of segments is allowed to change
dynamically with auto-scaling. Streams are grouped into scopes to avoid name collisions across
applications. 

One of the main APIs that Pravega provides is the event one. Events are serialized into bytes and
appended to segments. Pravega uses *routing keys* to map events to segments, they are
essentially sharded across open segments, and Pravega clients learn about open segments from
the control plane of Pravega.

Applications, including Flink jobs, read from a Pravega stream using reader groups. The readers
in a group share the workload of segments and coordinate the assignment of segments using a
Pravega feature that enables the coordination of state called *state synchronizer*. That coordination
is transparent to the application so that application does not have to handle dynamic changes to
the set of segments. This coordination also enables the generation of *checkpoints*, which are
essential to enable exactly-once semantics end-to-end with Flink. 

The foundation of checkpoints is an abstraction called *stream cut*: a collection of offsets spanning one or more segments. Checkpointing constitutes one of the main use cases of stream cuts, but stream cuts can additionally be used to set a position to begin reading or a position to end reading in a stream. To be able to resume or bound the computation over a stream of events, the Pravega API enables readers to obtain *stream cut* objects independent of checkpoints. 
```

Reply:
Replaced.